File: starter-code/Balance.java
import java.util.*;
public class Balance{
	private static int calls = 0, limit = 2;
	public static int[] generate(int[] balls) {
		Random rand = new Random();
		int genuine_weight = rand.nextInt(6)+19;
		int defective_weight = rand.nextInt(6)+30;
		int defective_position = rand.nextInt(balls.length);
		for (int i = 0; i < balls.length; i++){
			if (i != defective_position )
				balls[i] = genuine_weight;
		}
		balls[defective_position] = defective_weight;
		return balls;
	}
	public static int balance(int[] balls, Map<String, Integer> left, Map<String, Integer> right) {
		if (calls == limit){
        	System.out.println("The program is quitting because balance is called more than " + limit + " times ...");
        	System.exit(0);
        }
	    else
	        calls += 1;
	    
	    int lsum = 0;
	    for(int i = left.get("start"); i < left.get("end")+1; i++)
	    	lsum += balls[i];
	    
	    int rsum = 0;
	    for(int i = right.get("start"); i < right.get("end")+1; i++)
	        rsum += balls[i];

	    if (lsum == rsum)
	        return 0;
	    else if (lsum > rsum)
	        return -1;
	    else
	        return 1;
	}
	public static int puzzle(int[] balls){
		int result = 0;
		// add your logic here ....
		return result;
	}
	public static void main(String[] args) {
		int[] ball_weights = new int[9]; // a set of coins and their respective weights.
		/* does a random distribution of weights, while still
		making sure to be within the scope of the problem definition,
		that is, all values except one will be the same.
		the number of balls is fixed to 9.
		*/
		int[] balls = generate(ball_weights);
		System.out.println(Arrays.toString(balls));	
		System.out.println(puzzle(balls));
    	
		
	
	}	
}

File: starter-code/Balance.py
import sys
import random
limit,calls = 2,0
# This method is used to generate a list with random values.
def generate(balls,size):
    maximum = 25; # maximum ball weight 25 oz, assuming weight distributed between 19 and 25
    genuine_weight = random.randint(20,maximum)
    defective_weight = random.randint(30,genuine_weight+30)
    if (genuine_weight == defective_weight):
        defective_weight = genuine_weight - 1
    defective_position = random.randint(0, size-1);
    for i in range(0,size):
        if (i != defective_position):
            balls.insert(i,genuine_weight)
    balls.insert(defective_position,defective_weight)

def balance(balls,left,right):  
    global calls
    if (calls == limit):
        print(f"The program is quitting because balance is called more than {limit} times ...")
        sys.exit(0)
    else:
        calls += 1
    lsum = 0
    for i in range(left.get("start"),left.get("end")+1):
        lsum += balls[i]
    rsum = 0
    for i in range(right.get("start"),right.get("end")+1):
        rsum += balls[i]
    if (lsum == rsum):    
        return 0
    elif (lsum > rsum):
        return -1
    else:
        return 1

def puzzle(balls):
    result = 0
    # add your logic here by removing pass below ...
    return result

balls = []
size = 9
generate(balls,size)
print(balls)
result = puzzle(balls)
print(result)
balls = []